src/blueprint/operators/call.rs

```diff
@@ -1,152 +1,135 @@
-use std::collections::btree_map::Iter;
-
 use serde_json::Value;
 
 use crate::blueprint::*;
 use crate::config;
-use crate::config::{Field, GraphQLOperationType, KeyValue};
+use crate::config::{Field, GraphQLOperationType};
 use crate::lambda::Expression;
-use crate::mustache::{Mustache, Segment};
 use crate::try_fold::TryFold;
 use crate::valid::{Valid, ValidationError, Validator};
 
-fn find_value<'a>(args: &'a Iter<'a, String, Value>, key: &'a String) -> Option<&'a Value> {
-    args.clone()
-        .find_map(|(k, value)| if k == key { Some(value) } else { None })
-}
-
-pub fn update_call(
-    operation_type: &GraphQLOperationType,
-) -> TryFold<'_, (&ConfigModule, &Field, &config::Type, &str), FieldDefinition, String> {
+pub fn update_call<'a>(
+    operation_type: &'a GraphQLOperationType,
+    object_name: &'a str,
+) -> TryFold<'a, (&'a ConfigModule, &'a Field, &'a config::Type, &'a str), FieldDefinition, String>
+{
     TryFold::<(&ConfigModule, &Field, &config::Type, &str), FieldDefinition, String>::new(
-        move |(config, field, _, _), b_field| {
+        move |(config, field, _, name), b_field| {
             let Some(ref calls) = field.call else {
                 return Valid::succeed(b_field);
             };
 
-            compile_call(field, config, calls, operation_type)
-                .map(|resolver| b_field.resolver(Some(resolver)))
+            compile_call(config, calls, operation_type, object_name).map(|compiled| {
+                b_field
+                    .args(
+                        compiled
+                            .args
+                            .into_iter()
+                            .map(|mut input_value| {
+                                input_value.of_type = match &input_value.of_type {
+                                    Type::NamedType { name, .. } => {
+                                        Type::NamedType { name: name.to_owned(), non_null: false }
+                                    }
+                                    Type::ListType { of_type, .. } => Type::ListType {
+                                        of_type: of_type.to_owned(),
+                                        non_null: false,
+                                    },
+                                };
+
+                                input_value
+                            })
```

Comment:
That's needed because `blueprint/mustache.rs` validation requires the `b_field.args` to exist, but we set everything as non required, as the proper arg is being passed via `args` key on `@call`, and is checking to meet the args requirement on [these lines](https://github.com/tailcallhq/tailcall/pull/1632/files#diff-738fcff964193a22920814b6f5ea2d5959148d8a50929e61564da8863e2a9918R59-R81)

Reply:
That's the reason why the client snapshots were updated with some args.